parlai/core/logging_utils.py

```diff
@@ -12,32 +12,74 @@
 
 
 class ParlaiLogger(logging.Logger):
-    def __init__(self, name, filename, level, format):
-        super().__init__(name, level)
-        # Setting these here so that never have to import logging everywhere
-        self.NOTSET = logging.NOTSET
-        self.DEBUG = logging.DEBUG
-        self.INFO = logging.INFO
-        self.WARNING = logging.WARNING
-        self.ERROR = logging.ERROR
-        self.CRITICAL = logging.CRITICAL
+    def __init__(
+        self,
+        name,
+        console_level,
+        console_format=None,
+        file_format=None,
+        file_level=logging.INFO,
+        filename=None,
+    ):
+        """
+        Initialize the logger object
+        :param name: Name of the logger
+        :param console_level: min. Level of messages logged to console
+        :param console_format: The format of messages logget to the console. Simple stdout is used if None specified
+        :param file_format: The format of messages logged to the file (A default is used if None specified)
+        :param file_level: min. Level of messages logged to the file
+        :param filename: The file the logs are written to
+
+        A default file is initialized if no filename was provided. It can be
+        found in ParlAI/Logs/*year*/*month*/*YYYY-MM-DD*.log
+        """
+        # Setup logging location if none specified
+        if filename is None:
+            today = date.today().strftime("%d/%m/%Y").split("/")
+            day, month, year = today[0], today[1], today[2]
+            proj_root = str(Path(__file__).parent.parent.parent)
+            log_dir_path = proj_root + "/" + "Logs/" + year + "/" + month
+            log_fname = year + "-" + month + "-" + day + ".log"
+            if not os.path.exists(log_dir_path):
+                os.makedirs(log_dir_path)
+            filename = os.path.join(proj_root, log_dir_path, log_fname)
+        # Default format used if no file format provided
+        if file_format is None:
+            file_format = '%(asctime)s :: %(pathname)s :: %(funcName)s :: %(lineno)d :: %(levelname)s :: %(message)s'
```

Comment:
this `file_format` might be a little bit verbose, could we possibly have the default include less?

or, perhaps we can have an optional `verbose` flag for each log function that makes the prefix like this.

Reply:
The flag idea might not work since to implement that we need to write functions such as log,info,debug etc in the ParlaiLogger class and when we do that it is saying that we call those functions from within this file, thus making tracking the pathname, funcname and linno of no use since it will always be the same (wherever it is called in logging.utils). But I can reduce the default format to just ```time :: level ::  message```